gui/ui/src/component/form.rs

```diff
@@ -62,6 +64,29 @@ where
         }
     }
 
+    /// Creates a new [`Form`] that restrict input values to valid btc amount before applying the
+    /// `on_change` function.
+    /// It expects:
+    /// - a placeholder
+    /// - the current value
+    /// - a function that produces a message when the [`Form`] changes
+    pub fn new_amount_btc<F>(placeholder: &str, value: &'a Value<String>, on_change: F) -> Self
+    where
+        F: 'static + Fn(String) -> Message,
+    {
+        Self {
+            input: text_input::TextInput::new(placeholder, &value.value).on_input(move |s| {
+                if bitcoin::Amount::from_str_in(&s, Denomination::Bitcoin).is_ok() {
```

Comment:
I think it would also be good to allow an empty string so that a user can delete the value completely and start again.